### DISCUSSION TITLE
Can't use auth middleware

### DISCUSSION THREAD

**Discussion by D3xime** [adonisjs/core#4668] [Help]
Hello, 

I am new to Adonis framework, but i have some problems, i have followed the documentation and encounter a problem with the auth middleware. 

router
  .post('projects', async ({ auth }) => {
    console.log(auth.user) // User
    console.log(auth.authenticatedViaGuard) // 'api'
    console.log(auth.user!.currentAccessToken) // AccessToken
  })
  .use(
    middleware.auth({
      guards: ['api']
    })
  )
    [Capture d’écran 2024-07-26 à 11 37 38](https://github.com/user-attachments/assets/3837fbbc-569a-4cf1-9e28-933744b1a995)
Type script is advertising me about i can affect string to type never.
how can i achieve that ? 

here is my auth middleware 
import type { HttpContext } from '@adonisjs/core/http'
import type { NextFn } from '@adonisjs/core/types/http'
import type { Authenticators } from '@adonisjs/auth/types'

/**
 * Auth middleware is used authenticate HTTP requests and deny
 * access to unauthenticated users.
 */
export default class AuthMiddleware {
  /**
   * The URL to redirect to, when authentication fails
   */
  redirectTo = '/login'

  async handle(
    ctx: HttpContext,
    next: NextFn,
    options: {
      guards?: (keyof Authenticators)[]
    } = {}
  ) {
    await ctx.auth.authenticateUsing(options.guards, { loginRoute: this.redirectTo })
    return next()
  }
}

Thanks in advance for your help.

**Comment by dtmonterrey** [comment]
Make sure your middleware flow is correctly stacked. Check this guide: https://adocasts.com/lessons/the-flow-of-middleware
